cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2146,6 +2147,203 @@ class TakeMetaFunction : public MetaFunction {
   }
 };
 
+// ----------------------------------------------------------------------
+// DropNull Implementation
+
+Status GetDropNullFilter(const Array& values, MemoryPool* memory_pool,
+                         std::shared_ptr<arrow::BooleanArray>* out_array) {
+  auto bitmap_buffer = values.null_bitmap();
+  *out_array = std::make_shared<BooleanArray>(values.length(), bitmap_buffer, nullptr, 0,
+                                              values.offset());
+  return Status::OK();
+}
+
+Status CreateEmptyArray(std::shared_ptr<DataType> type, MemoryPool* memory_pool,
+                        std::shared_ptr<Array>* output_array) {
+  std::unique_ptr<ArrayBuilder> builder;
+  RETURN_NOT_OK(MakeBuilder(memory_pool, type, &builder));
+  RETURN_NOT_OK(builder->Resize(0));
+  ARROW_ASSIGN_OR_RAISE(*output_array, builder->Finish());
+  return Status::OK();
+}
```

Comment:
This seems like a very useful utility function.  I wonder if we want to put it somewhere more visible.  There could be a CreateEmptyRecordBatch and CreateEmptyTable too.  Can be done in a follow-up PR.

Reply:
Yep! We just had this discussion in Zulip. 
https://ursalabs.zulipchat.com/#narrow/stream/271283-help.2Fc.2B.2B/topic/creating.20empty.20array.2F.20table.20using.20Slice

https://issues.apache.org/jira/browse/ARROW-7051

Reply:
Yeap, I will create an issue on Jira

Reply:
https://issues.apache.org/jira/browse/ARROW-13579

Reply:
Nice!